src/inspect_ai/util/_sandbox/self_check.py

```diff
@@ -176,6 +178,39 @@ async def test_exec_permission_error(sandbox_env: SandboxEnvironment) -> None:
         await sandbox_env.exec(["/etc/passwd"])
 
 
+async def test_exec_as_user(sandbox_env: SandboxEnvironment) -> None:
+    try:
+        # Create a new user
+        add_user_result = await sandbox_env.exec(
+            ["useradd", "-m", "myuser"], user="root"
+        )
+        assert add_user_result.success, f"Failed to add user: {add_user_result.stderr}"
+
+        # Test exec as different users
+        root_result = await sandbox_env.exec(["whoami"], user="root")
+        assert (
+            root_result.stdout.strip() == "root"
+        ), f"Expected 'root', got '{root_result.stdout.strip()}'"
+        myuser_result = await sandbox_env.exec(["whoami"], user="myuser")
+        assert (
+            myuser_result.stdout.strip() == "myuser"
+        ), f"Expected 'myuser', got '{myuser_result.stdout.strip()}'"
+    finally:
+        # Clean up
+        await sandbox_env.exec(["userdel", "-r", "myuser"], user="root")
```

Comment:
Note: it seems like tests all reuse the same sandbox. The test I have introduced will add a user `myuser` to the sandbox and attempt to delete it at the end. But there could still be side-effects if the cleanup step fails. Possible alternatives:
* Keep my test the way it is and hope that it cleans up successfully
* Maybe introduce some more reliable cleanup? (Not sure how, would need to assert that the user has indeed been deleted before continuing -- could check `/etc/passwd`)
    * New sandbox for each test -- this will be costly for performance
    * Add the `myuser` user to the sandbox for ALL tests -- though this would mess with the local system for local sandbox.

Reply:
I think it's fine to rely on the finally here. If you wanted to do more verification you could add another test right after this one that assert that there is no such user (or perhaps the non-existent user test could use "myuser", is that too cute?).